backend/services/create_battle.py

```diff
@@ -0,0 +1,36 @@
+from users.models import User
+from django.utils.crypto import get_random_string
+from django.contrib.auth.forms import PasswordResetForm
+from battle.battles.email import send_invite_email
+from django.conf import settings
+
+
+def validate_if_creator_and_opponent_are_different(creator, opponent):
+    if creator == opponent:
+        return False
+    return True
+
+
+def validate_if_opponent_is_valid(opponent_email):
```

Comment:
The naming here could be improved, this is not a validation by itself, you are fetching an opponent or creating one if it doesn't exists

Reply:
hmm that's true, thanks!